src/transaction.ts

```diff
@@ -356,4 +342,68 @@ export default class Transaction {
     // Note: This never gets executed, defineProperties overwrites it.
     return rlp.encode(this.raw)
   }
+
+  private _validateV(v?: Buffer): void {
+    if (v === undefined || v.length === 0) {
+      return
+    }
+
+    if (!this._common.gteHardfork('spuriousDragon')) {
+      return
+    }
+
+    const vInt = bufferToInt(v)
+
+    if (vInt === 27 || vInt === 28) {
+      return
+    }
+
+    const isValidEIP155V =
+      vInt === this.getChainId() * 2 + 35 || vInt === this.getChainId() * 2 + 36
+
+    if (!isValidEIP155V) {
+      throw new Error(
+        `Incompatible EIP155-based V ${vInt} and chain id ${this.getChainId()}. See the second parameter of the Transaction constructor to set the chain id.`,
+      )
+    }
+  }
```

Comment:
Cautiously think this looks good.

Reply:
I thought a lot about this. I added it because the second param of the constructor is not that discoverable. 

If you aren't aware of it, and guided by the EIP155's text, you set a correct signature for other chain after creation, it would be considered invalid. That's one of the things that #152 describes.

Reply:
Ok, thanks for all the thought you put in here, that's a great explanation respectively reasoning.